### ISSUE TITLE
HTML macro tooltips can cause exception to be display

### ISSUE THREAD

**Issue by kwvanderlinde** [RPTools/maptool#2754]
**Describe the bug**

When hovering over a macro button with an HTML tooltip, an exception is sometimes displayed in a warning dialog.

**To Reproduce**
1. Drop a token and create a macro on it with an HTML tooltip.
2. Hover over the macro to reveal a tooltip. Might take a few tries.
3. If successful, there will be a dialog titled "Warning" with an exception stack trace, though the tooltip is also revealed.

**Expected behavior**
The tooltip should be displayed without the warning dialog showing.

**Screenshots**
![tooltip error](https://user-images.githubusercontent.com/7492219/122603266-ee499100-d028-11eb-97c7-2de4cc2a6e18.png)

**MapTool Info**
- Version: 1.9.0
- Install: Upgrade from 1.8.3

**Desktop (please complete the following information):**
 - OS: Linux Mint
 - Version 20

**Additional context**
Unfortunately this isn't entirely consistent. Sometimes it happens on every tooltip reveal, other times I see no issue for a long time, and sometimes it's just hit-or-miss. As far as I'm aware, there isn't anything different about each situation.

Here is the message from one of the exceptions:
<details>
<summary>

```
java.lang.IllegalArgumentException: Width and height must be >= 0
	at java.desktop/javax.swing.plaf.basic.BasicHTML.getHTMLBaseline(Unknown Source)
	at java.desktop/javax.swing.plaf.metal.MetalToolTipUI.paint(Unknown Source)
	at java.desktop/javax.swing.plaf.ComponentUI.update(Unknown Source)
```
</summary>

```
	at java.desktop/javax.swing.JComponent.paintComponent(Unknown Source)
	at java.desktop/javax.swing.JComponent.paint(Unknown Source)
	at java.desktop/javax.swing.JComponent.paintChildren(Unknown Source)
	at java.desktop/javax.swing.JComponent.paint(Unknown Source)
	at java.desktop/javax.swing.JComponent.paintChildren(Unknown Source)
	at java.desktop/javax.swing.JComponent.paint(Unknown Source)
	at java.desktop/javax.swing.JLayeredPane.paint(Unknown Source)
	at java.desktop/javax.swing.JComponent.paintChildren(Unknown Source)
	at java.desktop/javax.swing.JComponent.paintToOffscreen(Unknown Source)
	at java.desktop/javax.swing.RepaintManager$PaintManager.paintDoubleBufferedImpl(Unknown Source)
	at java.desktop/javax.swing.RepaintManager$PaintManager.paintDoubleBuffered(Unknown Source)
	at java.desktop/javax.swing.RepaintManager$PaintManager.paint(Unknown Source)
	at java.desktop/javax.swing.BufferStrategyPaintManager.paint(Unknown Source)
	at java.desktop/javax.swing.RepaintManager.paint(Unknown Source)
	at java.desktop/javax.swing.JComponent.paint(Unknown Source)
	at java.desktop/java.awt.GraphicsCallback$PaintCallback.run(Unknown Source)
	at java.desktop/sun.awt.SunGraphicsCallback.runOneComponent(Unknown Source)
	at java.desktop/sun.awt.SunGraphicsCallback.runComponents(Unknown Source)
	at java.desktop/java.awt.Container.paint(Unknown Source)
	at java.desktop/java.awt.Window.paint(Unknown Source)
	at java.desktop/javax.swing.RepaintManager$4.run(Unknown Source)
	at java.desktop/javax.swing.RepaintManager$4.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.desktop/javax.swing.RepaintManager.paintDirtyRegions(Unknown Source)
	at java.desktop/javax.swing.RepaintManager.paintDirtyRegions(Unknown Source)
	at java.desktop/javax.swing.RepaintManager.prePaintDirtyRegions(Unknown Source)
	at java.desktop/javax.swing.RepaintManager$ProcessingRunnable.run(Unknown Source)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(Unknown Source)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(Unknown Source)
	at java.desktop/java.awt.EventQueue$4.run(Unknown Source)
	at java.desktop/java.awt.EventQueue$4.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.desktop/java.awt.EventQueue.dispatchEvent(Unknown Source)
	at net.rptools.maptool.client.swing.MapToolEventQueue.dispatchEvent(MapToolEventQueue.java:54)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.desktop/java.awt.WaitDispatchSupport$2.run(Unknown Source)
	at java.desktop/java.awt.WaitDispatchSupport$4.run(Unknown Source)
	at java.desktop/java.awt.WaitDispatchSupport$4.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at java.desktop/java.awt.WaitDispatchSupport.enter(Unknown Source)
	at java.desktop/java.awt.Dialog.show(Unknown Source)
	at java.desktop/java.awt.Component.show(Unknown Source)
	at java.desktop/java.awt.Component.setVisible(Unknown Source)
	at java.desktop/java.awt.Window.setVisible(Unknown Source)
	at java.desktop/java.awt.Dialog.setVisible(Unknown Source)
	at net.rptools.maptool.client.swing.MapToolEventQueue.displayPopup(MapToolEventQueue.java:104)
	at net.rptools.maptool.client.swing.MapToolEventQueue.dispatchEvent(MapToolEventQueue.java:68)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.run(Unknown Source)
```
</details>

**Comment by kwvanderlinde** [comment]
Google showed me a [JSOM bug ticket](https://josm.openstreetmap.de/ticket/19585) from which I gleaned some information:
- It's been fixed in Java 17: [JDK-8262085](https://bugs.openjdk.java.net/browse/JDK-8262085)
- It only affects Metal L&F (not surprising given the stack trace)
- It apparently only affects certain linux distros with OpenJDK.
- Debian has backported a fix to Java 11. I guess AdoptOpenJDK doesn't have a similar backport for Java 16.

**Comment by Phergus** [comment]
That perhaps explains why I couldn't get it to happen.
